File: estateapp/estateapp/report/property/property.js
// Copyright (c) 2022, test and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Property"] = {
	"filters": [
		{
			"fieldname":"property",
			"label":__("Property Name"),
			"fieldtype":"Data",
			"width": 100,
			"reqd":0
		},
		{
			"fieldname":"from",
			"label":__("From Date"),
			"fieldtype":"Date",
			"width": 100,
			"reqd":1,
			"default":frappe.datetime.year_start()
		},
		{
			"fieldname":"to",
			"label":__("To Date"),
			"fieldtype":"Date",
			"width": 100,
			"reqd":1,
			"default":frappe.datetime.year_end()
		},
		{
			"fieldname":"agent",
			"label":__("Agent Name"),
			"fieldtype":"Link",
			"options": "Agent",
			"width": 100,
			"reqd":0
		}
	]
};

File: estateapp/estateapp/report/property/property.py
# Copyright (c) 2022, test and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	return get_columns(), get_data(filters)

def get_columns():
	return [
		"ID:Link/Property:70",
		"Property Name:Data:150",
		"Type:Data:80",
		"Agent:Data:150",
		"Status:Select:70",
	]

def get_data(filters):
	_from = filters["from"]
	to = filters["to"]
	conditions = ''
	if (filters.get('property')): conditions += f" AND property_name LIKE '%{filters.get('property')}%'"
	if (filters.get('agent')): conditions += f" AND agent='{filters.get('agent')}'"
	list = frappe.db.sql(f"""SELECT name,property_name,property_type,agent,status FROM tabProperty WHERE (creation BETWEEN '{_from}' AND '{to}') {conditions}""")
	return list
